### PR TITLE
ARGO-398 Add Ack Request and Align Error Messages

### PR DESCRIPTION
**Pull request by kaggis**
- [x] Add Optional Ack request. With Ack request subscriber can validate that he indeed received the messages and API cant progress to the next offset. If Ack is turned off in confuguration progression is automatic during consumption
- [x]  Revisit the set of available Error Messages and try to make them consistent in each case
- [x] Minor fixes, refactorings

### CODE REVIEW

**Comment by kaggis** [comment]
please review
/cc @pkoro

File: config/config_test.go [hunk 1]
```diff
@@ -14,47 +14,53 @@ type ConfigTestSuite struct {
 }
 
 func (suite *ConfigTestSuite) SetupTest() {
-	suite.cfgStr = `
-	{
-	  "broker_host":"localhost:9092",
+	suite.cfgStr = `{
+	  "port":8080,
```
Comment (pkoro) [inline config/config_test.go]: There is a minor indentation issue here.

File: handlers_test.go [hunk 2]
```diff
@@ -697,6 +658,86 @@ func (suite *HandlerTestSuite) TestSubPullAll() {
...
+	str.SubList[0].NextOffset = 4
+
+	req3, err := http.NewRequest("POST", url, bytes.NewBuffer([]byte(postJSON3)))
+	router3 := mux.NewRouter().StrictSlash(true)
+	w3 := httptest.NewRecorder()
+	router3.HandleFunc("/v1/projects/{project}/subscriptions/{subscription}:acknowledge", WrapConfig(SubAck, cfgKafka, &brk, str))
+	router3.ServeHTTP(w3, req3)
+	suite.Equal(408, w3.Code)
+	suite.Equal(expJSON2, w3.Body.String())
+	fmt.Println(w3.Body.String())
```
Comment (pkoro) [inline handlers_test.go]: Remove this line

File: stores/mongo.go [hunk 3]
```diff
@@ -204,6 +277,7 @@ func (mong *MongoStore) InsertResource(col string, res interface{}) error {
 
 // RemoveResource removes a resource from the store
 func (mong *MongoStore) RemoveResource(col string, res interface{}) error {
+	fmt.Println(res)
```
Comment (pkoro) [inline stores/mongo.go]: Remove this line

**Comment by pkoro** [comment]
+1

**Pull request merged**